src/python/pants/engine/isolated_process.py

```diff
@@ -138,16 +146,16 @@ def __init__(self, exit_code, stdout, stderr, process_description):
     super().__init__(msg)
 
 
-@rule(ProductDescription, [ExecuteProcessRequest])
-def get_request_description(req):
-  return ProductDescription(req.description)
-
-
 @rule(ProductDescription, [MultiPlatformExecuteProcessRequest])
 def get_multi_platform_request_description(req):
-  # we can safely extract the first description because we guarantee that at
-  # least one request exists and that all of their descriptions are the same.
-  return ProductDescription(req.execute_process_requests[0].description)
+  return req.get_product_description()
+
+
+@rule(MultiPlatformExecuteProcessRequest, [ExecuteProcessRequest])
+def upcast_execute_process_request(req):
```

Comment:
A comment explaining this one would be helpful (enables executing a `ExecuteProcessRequest` directly by defaulting to cross platform execution...?).